lib/Models/Catalog/Ows/WebFeatureServiceCatalogItem.ts

```diff
@@ -332,21 +332,47 @@ class WebFeatureServiceCatalogItem extends GetCapabilitiesMixin(
     }
 
     // Check if geojson output is supported (by checking GetCapabilities OutputTypes OR FeatureType OutputTypes)
-    const hasOutputFormat = (outputFormats: string[] | undefined) => {
+    const hasJsonOutputFormat = (outputFormats: string[] | undefined) => {
       return isDefined(
         outputFormats?.find((format) =>
           ["json", "JSON", "application/json"].includes(format)
         )
       );
     };
 
+    // Find which GML formats are supported, choose the one most suited to Terria. If not available, default to "gml3"
+    const getBestGmlOutputFormat = () => {
+      const searchValue = new RegExp(".*gml/3.1.1.*|.*gml3.1.1.*");
+      return (
+        getCapabilitiesStratum.capabilities.outputTypes?.find((outputFormat) =>
+          searchValue.test(outputFormat)
+        ) ?? "gml3"
+      );
+    };
+
+    // Returns the first listed srs that contains `4326`. This enables us to use a urn identifier if supported, or a normal EPSG code if not.
+    // e.g. "urn:ogc:def:crs:EPSG::4326" or "EPSG:4326"
+    const getBestSrsName = (layerNamesArray: readonly string[]) => {
+      // Just use the first layer in the layerNames array.
+      // If in the rare case we are making a WFS request for multiple layers, we will assume that all layers use the same code for WGS84 coordinate system
+      const layerSrsArray = getCapabilitiesStratum.capabilities.srsNames?.find(
+        (layer) => layer.layerName === layerNamesArray[0]
+      );
+
+      const searchValue = new RegExp("4326");
+      const result =
+        layerSrsArray?.srsArray.find((srsName) => searchValue.test(srsName)) ??
+        "urn:ogc:def:crs:EPSG::4326"; // Default to urn identifier for WGS84 if we cant find something better.
+      return result;
+    };
+
```

Comment:
I think it would be better to create two new `Traits` in `WebFeatureServiceCatalogItemTraits` - something like `srsName` and `outputFormat`.

Then you can add default values for both in `GetCapabilitiesStratum` (in `lib/Models/Catalog/Ows/WebFeatureServiceCatalogItem.ts`). It would also simplify the `forceLoadGeojsonData` function - as you can move all of that logic into `GetCapabilitiesStratum`.

This would allow the values to be overridden in JSON catalog.

A similar example is how `crs` is handled in WMS:

Where we have a `crs` trait in `WebMapServiceCatalogItemTraits`

https://github.com/TerriaJS/terriajs/blob/588eb6a63bc1c23d1629d5fc0a4920ffff4b5411/lib/Traits/TraitsClasses/WebMapServiceCatalogItemTraits.ts#L195-L202

Which is given a default value in `WebMapServiceCapabilitiesStratum`

https://github.com/TerriaJS/terriajs/blob/cad62a45cbee98c7561625458bec3a48510f6cbc/lib/Models/Catalog/Ows/WebMapServiceCapabilitiesStratum.ts#L299-L314

Reply:
Great, thanks for the advice Nick! Just what I was looking for, will dig in in this direction.

Reply:
OK @nf-s Ive implemented this all in a simple neat way.
One thing I havent done, in contract to `WebMapServiceCatalogItem`, is to check for srsName in the provided url. I dont think this is important, we just override with the best one anyway.